internal/controller/components/kserve/kserve_controller.go

```diff
@@ -96,9 +97,11 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 				component.ForLabel(labels.ODH.Component(LegacyComponentName), labels.True),
 			),
 		).
+		Watches(&v1alpha1.Subscription{}).
```

Comment:
Adding predicate to the Watch here will limit the controller to only enter reconcile loop when there are changes to RHCL and LWS. This also requires adding `"sigs.k8s.io/controller-runtime/pkg/predicate"` as an import.

```suggestion
		Watches(&v1alpha1.Subscription{},
			reconciler.WithEventHandler(
				handlers.ToNamed(componentApi.KserveInstanceName),
			),
			reconciler.WithPredicates(
				predicate.Or(
					resources.CreatedOrUpdatedOrDeletedNamed(rhclOperator),
					resources.CreatedOrUpdatedOrDeletedNamed(lwsOperator),
				),
			),
		).

```